File: uva_11121.cpp
# include <vector>
# include <iostream>
# include <algorithm>

using namespace std;

vector<int> base_2(int num){
    vector<int> output;
    while(num){
        output.push_back(num%(-2));
        num/=(-2);
        if(output[output.size()-1]==-1){
            output[output.size()-1]*=(-1);
            ++num;
        }
    }
    if(output.empty())
        output.push_back(0);
    return output;
}

int main(void){
    int num=0;
    cin >> num;
    for (int index=0; index<num; ++index){
        int input=0;
        cin >> input;
        vector<int> output;
        output = base_2(input);
        cout << "Case #" << index+1 << ": ";
        for(int i=output.size()-1; i>=0; --i)
            cout << output[i];
        cout << endl;
    }
    return 0;
}

File: uva_11121.py
def base_2(num):
    output=[]
    while num:
        mod=num%(-2)
        if num>0 and mod<0:
            mod=mod+2
        output.append(mod)
        num=num/(-2)
        num=int(num)
        if output[-1]<0:
            output[-1]+=2
            num+=1
    if not len(output):
        output.append(0)
    return output

input_num=input()
input_num=int(input_num)
for i in range(input_num):
    data=input()
    data=int(data)
    output=base_2(data)
    output.reverse()
    print("Case #{}: ".format(i+1), end='')
    for item in output:
        print(item, end='')
    print('')
